keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
+    |      "bucketID": "br-$uuid",             |
+    |      "parameters": {                     |
+    |         "key": "value"                   |
+    |      }                                   |
+    | })                                       |
+    |------------------------------------------|
+```
 
-![CreateBucket Workflow](images/create-bucket.png)
+Each BucketClaim is meant to map to a unique service account in the OSP. Once the requested privileges have been granted, a secret by the name specified in `credentialsSecretName` in the BucketClaim is created. The secret will reside in the namespace of the BucketClaim. The secret will cotain either keys or service account mappings based on the chosen authentication type. 
 
-This workflow describes the automation supporting creating a new (greenfield) backend bucket. Accessing this bucket is covered in [Sharing COSI Created Buckets](#sharing-cosi-created-buckets). COSI determines that a request is for an existing bucket when `BucketRequest.bucketInstanceName` is specified.
+If this call returns successfully, the sidecar sets `AccessGranted` condition to `True` in the BucketAccess.
 
-Here is the workflow:
+NOTE:
+ - The secret will not be created until the credentials are generated/service account mappings are complete.
+ - Within a namespace, one BucketAccess and secret pair is generally sufficient, but cases which may want to control this more granularly can use multiple.
 
-+ COSI central controller sees a new `BucketRequest` (BR).
-+ COSI gets the `BR.BucketClass` (directly or via the matching default) and verifies that the BR's namespace is allowed.
-+ If BR.bucketInstanceName is empty, COSI creates the associated `Bucket`, filling in fields from the `BucketClass` and BR. Else, we have a brownfield BR.
-+ The sidecar sees the new `Bucket` and gRPC calls the driver to create a backend bucket.
-+ COSI updates the BR and `Bucket` instance to reference each other.
+## Attaching Buckets
 
-In addition, the cosi-node-adapter sees a new app pod references a BAR:
-+ cosi-node-adapter receives _NodeStageVolume_ request from kubelet.
-+ adapter add a finalizer to both the BAR.BA and the BAR.BR.
-+ the finalizer value is `pod.name`
+The following stakeholders are involved in the lifecycle of attaching bucket to pods:
 
-## Sharing COSI Created Buckets
-This is the greenfield -> brownfield access use case, where COSI has created the `Bucket` instance and the driver has provisioned a new bucket. Now, we want to share access to this bucket in other namespaces.
+ - Users  - specify bucket in the pod definition
 
-![ShareBucket Workflow](images/share-bucket.png)
+Once a valid BucketClaim is available (`AccessGranted` is `True`), pods can use them.
 
-If the bucket sharing is all within the same namespace then each `BucketAccessRequest` (in the namespace) only needs to reference the existing BR.
+###### 1. User creates pod with a projected volume pointing to the secret in BucketAccess
```

Comment:
Is projected volume required, or can env vars work as well?

Can you also give an example where a Pod has access to 2 buckets?

Reply:
EnvVars will also work if an application knows where to look. 

> Can you also give an example where a Pod has access to 2 buckets?
When migrating data from one bucket to another (say from S3 to MinIO)

Reply:
Yes, I'll add both